editoast/src/views/timetable/stdcm.rs

```diff
@@ -191,6 +184,43 @@ async fn stdcm(
     let earliest_departure_time = stdcm_request.get_earliest_departure_time(simulation_run_time);
     let latest_simulation_end = stdcm_request.get_latest_simulation_end(simulation_run_time);
 
+    let timetable = Timetable::retrieve_or_fail(&mut conn, timetable_id, || {
+        StdcmError::TimetableNotFound { timetable_id }
+    })
+    .await?;
+
+    // Filtering trains is done in 2 stages:
+    // 1) We filter all trains whose departure dates are before the latest_simulation_end via sql
+
+    let mut train_schedules = timetable
+        .schedules_in_time_window(&mut conn, latest_simulation_end)
+        .await?;
+
+    // 2) We then filter in rust according to the lmr train departure date.  Since TrainSchedule input is light, this rust filtering saves time.
```

Comment:
I think it would be more helpful to explain why we're doing this manipulation with `schedule` and `path`, and why the departure date isn't part of the SQL query. It'd be great if you could reproduce the schema you shown me on call in ascii-art, but at least explaining it would help better understand this part. (If you ask me again in six months I'm not I'd be able to tell why again...)

I'd also be cautious about "Since TrainSchedule input is light, this rust filtering saves time". We don't know that, we just assume it's good enough. I'd argue it's actually the opposite: rewriting the following filtering in SQL could be an optimization to bench for large grids, saving us transport cost.